porch/engine/pkg/kpt/fs_test.go

```diff
@@ -168,35 +192,63 @@ metadata:
     tier: db
 spec:
   image: nginx:1.2.3
+`
+	expectedDbResources := `apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: db
+  namespace: staging
+  labels:
+    app: backend
+    tier: db
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      app: backend
+      tier: db
+  template:
+    metadata:
+      labels:
+        app: backend
+        tier: db
 `
 	fs := filesys.MakeFsInMemory()
-	if err := fs.MkdirAll("app"); err != nil {
-		t.Errorf(`MkdirAll("a/b/c") failed %v`, err)
+	if err := fs.MkdirAll("app/db"); err != nil {
+		t.Errorf(`MkdirAll("app/db") failed %v`, err)
 	}
 	if err := fs.WriteFile("/app/resources.yaml", []byte(appResources)); err != nil {
 		t.Errorf("Failed to write file: %v", err)
 	}
 	if err := fs.WriteFile("/app/Kptfile", []byte(appKptfile)); err != nil {
 		t.Errorf("Failed to write file: %v", err)
 	}
+	if err := fs.WriteFile("/app/db/resources.yaml", []byte(dbResources)); err != nil {
+		t.Errorf("Failed to write file: %v", err)
+	}
+	if err := fs.WriteFile("/app/db/Kptfile", []byte(dbKptfile)); err != nil {
+		t.Errorf("Failed to write file: %v", err)
+	}
 
 	r := render.Renderer{
 		PkgPath:         "/app",
 		ImagePullPolicy: fnruntime.IfNotPresentPull,
 		FileSystem:      fs,
-		Runtime:         &runtime{},
```

Comment:
let's not remove this; I'd rather have unit tests not depend on Docker.

Reply:
Updated.